pkg/provider/azure_loadbalancer.go

```diff
@@ -452,7 +421,8 @@ func (az *Cloud) cleanOrphanedLoadBalancer(lb *network.LoadBalancer, existingLBs
 	serviceName := getServiceName(service)
 	isBackendPoolPreConfigured := az.isBackendPoolPreConfigured(service)
 	lbResourceGroup := az.getLoadBalancerResourceGroup()
-	lbBackendPoolName := getBackendPoolName(clusterName, service)
+	isIPv6 := utilnet.IsIPv6String(service.Spec.ClusterIP)
+	lbBackendPoolName := getBackendPoolName(clusterName, isIPv6)
```

Comment:
What if there are both ipv4 and v6 addresses on a service?

Reply:
This is not the final code. Changes in `azure_loadbalancer.go` is still following single stack design. Will update in the next PR.

Reply:
Correct me if I am wrong. It is expected to have two backend pools when creating a dual-stack service with both ipv4 and v6 addresses. But this change only check one of the address so that it doesn't support reconciling two backend pools at a time.

Reply:
Indeed. Such feature still needs changes in `azure_loadbalancer.go` and to keep the PR size reasonable (current one is still larget actually), I will put these changes in another PR.
So, in the title, I emphasize changes is only in azure_loadbalancer_backendpool.go and related files.

Reply:
Currently, the CCM behavior is still the same with before. No dualstack support.

Reply:
I edit the description for you, please check. Since this change is transparent to the user, I remove the release note.

Reply:
Thank you! It looks good.